lib/rules/no-restricted-imports.js

```diff
@@ -0,0 +1,46 @@
+/**
+ * @fileoverview Restrict usage of specified node imports.
+ * @author Guy Ellis
+ */
+"use strict";
+
+//------------------------------------------------------------------------------
+// Rule Definition
+//------------------------------------------------------------------------------
+
+module.exports = function(context) {
+    var restrictedImports = context.options;
+
+    // if no imports are restricted we don"t need to check
+    if (restrictedImports.length === 0) {
+        return {};
+    }
+
+    return {
+        "ImportDeclaration": function(node) {
+            if (node && node.source && node.source.value) {
+
+                var value = node.source.value.trim();
+
+                if (restrictedImports.indexOf(value) !== -1) {
```

Comment:
I just added this test locally: `{ code: "import os from \"os\";", options: ["osx"], ecmaFeatures: { modules: true } },` and it passed. If I understood you correctly then this should have failed right? I'll push this shortly when I update the other items you pointed out.